Surface_mesher/demo/Surface_mesher/volume.cpp

```diff
@@ -1518,8 +1515,26 @@ void Volume::gl_draw_marchingcube()
       list_draw_marching_cube_is_valid = (::glGetError() == GL_NO_ERROR);
     }
     if(!list_draw_marching_cube_is_valid)
-      std::cerr << boost::format("OpenGL error: %1%\n") 
-        % ::gluErrorString(::glGetError());
+    {
+       GLenum error = ::glGetError();
+      if(error != GL_NO_ERROR)
+      {
+        if(error == GL_INVALID_ENUM)
+          std::cerr << "An unacceptable value is specified for an enumerated argument." << "@" << line << std::endl;
+        if(error == GL_INVALID_VALUE)
+          std::cerr << "A numeric argument is out of range." << "@" << line << std::endl;
+        if(error == GL_INVALID_OPERATION)
+          std::cerr << "The specified operation is not allowed in the current state." << "@" << line << std::endl;
+        if(error == GL_INVALID_FRAMEBUFFER_OPERATION)
+          std::cerr << "The framebuffer object is not complete." << "@" << line << std::endl;
+        if(error == GL_OUT_OF_MEMORY)
+          std::cerr << "There is not enough memory left to execute the command." << "@" << line << std::endl;
+        if(error == GL_STACK_UNDERFLOW)
+          std::cerr << "An attempt has been made to perform an operation that would cause an internal stack to underflow." << "@" << line << std::endl;
+        if(error == GL_STACK_OVERFLOW)
+          std::cerr << "An attempt has been made to perform an operation that would cause an internal stack to overflow." << "@" << line << std::endl;
+      }
+    }
```

Comment:
I have already seen that piece of code in a recent patch you wrote.

Instead of a copy-paste, that would be better to create a re-usable function somewhere in CGAL.

[According to the specification][doc], `glGetError` must be called in a loop. There was a function called `gl_check_error`, in the polyhedron demo, but [it was removed][c]. I think it would be a good to reintroduce it, or something similar.

[doc]: https://www.opengl.org/sdk/docs/man/docbook4/xhtml/glGetError.xml
[c]: https://github.com/CGAL/cgal/commit/fbeb14e5d9e14195ed430723533e93508ea51f5f#diff-1327d860c5cfb1ce1642cd24f94c7020L28